File: geekbang/algorithm/source-code/1/lc206 反转链表.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
 var reverseList = function(head) {
    let now = head;
    let last = null;
    // 遍历
    while (now != null) {
        // 备份now.next
        let next = now.next;
        // 操作：now指向last
        now.next = last;
        // 往后移动
        last = now;
        now = next;
    }
    return last;
};

File: geekbang/algorithm/source-code/1/lc206 反转链表.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* now = head;
        ListNode* last = nullptr;
        // 遍历
        while (now != nullptr) {
            // 备份now.next
            ListNode* next = now->next;
            // 操作：now指向last
            now->next = last;
            // 往后移动
            last = now;
            now = next;
        }
        return last;
    }
};